### PR TITLE
[k8s] Update library, allow deployment API to fail

### PR DESCRIPTION
**Pull request by conorbranagan**
### What does this PR do?

Update k8s library, allow deployment API to fail.

* The latest version of the library includes better logging for errors
which include the status code. See https://github.com/ericchiang/k8s/pull/58
* The deployment API is unavailable in older versions of Kubernetes. We
don't really _need_ this information since we can parse it from the
ReplicaSet name.

### Motivation

We've had a couple users (https://trello.com/c/5VjK5smo/560-braintree-having-trouble-with-custom-api-server-port-for-kubernetes, https://trello.com/c/gMllhQNi/605-k8-side-bar-filters) who hit an error in deployments API. Right now the first case at least ended up being an issue with missing RBAC configuration. The update to the library should provide some additional context here.

### CODE REVIEW

**Review by masci — APPROVED** [review]
LGTM!

**Comment by masci** [comment]
Tests are failing for a known issue, a fix is on the way

**Review by hkaj — DISMISSED** [review]
makes sense 👍

File: pkg/metadata/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -53,7 +53,10 @@ func GetPayload() (metadata.Payload, error) {
 	if ok && apiErr.Code == http.StatusNotFound {
 		dr = &appsv1beta1.DeploymentList{}
 	} else if err != nil {
-		return nil, fmt.Errorf("failed to get deployments: %s", err)
+		// Allow Deployments API to fail, it's not available in all version and we
+		// can also parse this data from the replica-set name on the backend.
+		log.Warnf("Failed to retrieve Kubernetes deployments: %s", err)
```
Comment (masci) [inline pkg/metadata/kubernetes/kubernetes.go]: didn't notice but `log` package was not imported

**Review by masci — CHANGES_REQUESTED** [review]

File: pkg/metadata/kubernetes/kubernetes.go (hunk 1 above)
Comment (conorbranagan) [reply pkg/metadata/kubernetes/kubernetes.go]: Doh!

**Review by masci — APPROVED** [review]

**Pull request merged**